packages/transaction-manager/src/inmemory-transaction-manager.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  DataAccessTypes,
+  DecryptionProviderTypes,
+  EncryptionTypes,
+  TransactionTypes,
+} from '@requestnetwork/types';
+
+import { EventEmitter } from 'events';
+
+import BaseTransactionManager from './base-transaction-manager';
+import processTransaction from './utils/process-transaction';
+
+/**
+ * Implementation of InMemoryTransactionManager layer without encryption
+ */
+export default class InMemoryTransactionManager extends BaseTransactionManager {
+  public constructor(
+    dataAccess: DataAccessTypes.IDataAccess,
+    decryptionProvider?: DecryptionProviderTypes.IDecryptionProvider,
+  ) {
+    super(dataAccess, decryptionProvider);
+  }
+
+  /**
+   * Creates and returns an in memory transaction without persisting it . If encryptionParams is given, the transaction will be encrypted
+   *
+   * @param transactionData transaction to persist
+   * @param channelId string to identify a group of transactions
+   * @param encryptionParams list of encryption parameters to encrypt the channel key with
+   * @param topics list of string to topic the transaction*
+   * @returns object containing the meta-data in-memory transaction
+   */
+  public async persistTransaction(
+    transactionData: TransactionTypes.ITransactionData,
+    channelId: string,
+    topics?: string[],
+    encryptionParams: EncryptionTypes.IEncryptionParameters[] = [],
+  ): Promise<TransactionTypes.IReturnPersistTransaction> {
+    const { channelEncryptionMethod } = await processTransaction({
+      transactionData,
+      channelId,
+      encryptionParams,
+      dataAccess: this.dataAccess,
+      channelParser: this.channelParser,
+    });
+
+    // Create the return result with EventEmitter
+    const result: TransactionTypes.IReturnPersistTransaction = Object.assign(new EventEmitter(), {
+      meta: {
+        encryptionMethod: channelEncryptionMethod,
+      },
```

Comment:
You might need to add the `persistTransaction.meta`. I'm not sure how other components will react if it's missing. Please investigate.

```suggestion
      meta: {
        dataAccessMeta: persistResult.meta,
        encryptionMethod: channelEncryptionMethod,
      },
```